blog/_posts/2019-11-13-affiliations-feature.md

```diff
@@ -0,0 +1,67 @@
+
+Cleaning the data turned out to be more of an art than a science.  There exist curation tools for cleaning up messy data - the one we used almost exclusively was Open Refine (called Google Refine at the time).  This allowed bulk substitutions, expansions, translations and more.  But cleaning is not enough.  For example, changing all instances of “UC Irvine” to “University of California Irvine” may seem like a good idea, except that new instances of “UC Irvine” are likely to continue coming in.
+		 	 	 		
+We were working with a dataset of approximately 34 million unique strings (about 4.5m astronomy and 29+ million physics). In the end, we were able to identify approximately 2.25 million affiliation strings, resulting in matching 93% of astronomy affiliations and 70% of physics affiliations to an institutional identifier.
+
+
+Automated Pipelines: assigning identifiers to specific author-affiliation-bibcode entities
+
+With the initial set of affiliation strings identified we, along with ADS backend programmer Stephen McDonald, designed an automated pipeline to add canonical affiliation identifiers to our database of 15 million references.  It’s a simple process.  First we normalize all of the strings with all uppercase letters, remove spaces and a subset of punctuation marks not needed for disambiguation, which results in a reduced number of matching strings; often, strings will be identical without punctuation, such as strings with “... U.S.A.” and “... USA”.  We then assign the strings to a dictionary, where the strings themselves are the keys, and the values are the identifiers.  If a normalized incoming string matches a key in the dictionary, we assign it the same aff_id, and the record with augmented affiliation data is sent back to our database of metadata.  The augmented affiliations are then passed to Solr and then to the ADS website with the next reindexing process.
```

Comment:
* "where the strings themselves are the keys, and the *identifiers* are the *values*." Flip around for a parallel construction.
* "If a normalized incoming string matches a key in the dictionary, we assign it the *appropriate/matching* aff_id,"
* "the record with augmented affiliation data is sent back to our database of metadata. The augmented affiliations are then passed to Solr and then to the ADS website with the next reindexing process." Can we just say that the record is updated with the augmented affiliation data? This might be too much detail for a casual reader unfamiliar with our data pipelines.

Reply:
make sure you like my change